api/src/pcapi/core/offerers/repository.py

```diff
@@ -72,16 +73,55 @@ def get_offer_counts_by_venue(venue_ids: Iterable[int]) -> dict[int, int]:
     Venues that do not have any offers are not included in the
     returned dictionary.
     """
-    return dict(
-        Offer.query.filter(
-            Offer.validation != OfferValidationStatus.DRAFT,
-            Offer.venueId.in_(venue_ids),
-        )
-        .with_entities(Offer.venueId, sqla.func.count())
-        .group_by(Offer.venueId)
-        .all()
+    is_new_model_enabled = FeatureToggle.ENABLE_NEW_COLLECTIVE_MODEL.is_active()
+
+    offer_query = Offer.query.filter(
+        Offer.validation != OfferValidationStatus.DRAFT,
+        Offer.venueId.in_(venue_ids),
     )
 
+    if is_new_model_enabled:
+        offer_query = offer_query.filter(Offer.isEducational.is_(False))
+
+    individual_offers_count = dict(
+        offer_query.with_entities(Offer.venueId, sqla.func.count()).group_by(Offer.venueId).all()
+    )
+
+    collective_offers_count = {}
+    collective_offers_template_count = {}
+
+    if is_new_model_enabled:
+        collective_offers_count = dict(
+            CollectiveOffer.query.filter(
+                CollectiveOffer.validation != OfferValidationStatus.DRAFT,
+                CollectiveOffer.venueId.in_(venue_ids),
+            )
+            .with_entities(CollectiveOffer.venueId, sqla.func.count())
+            .group_by(CollectiveOffer.venueId)
+            .all()
+        )
+
+        collective_offers_template_count = dict(
+            CollectiveOfferTemplate.query.filter(
+                CollectiveOfferTemplate.validation != OfferValidationStatus.DRAFT,
+                CollectiveOfferTemplate.venueId.in_(venue_ids),
+            )
+            .with_entities(CollectiveOfferTemplate.venueId, sqla.func.count())
+            .group_by(CollectiveOfferTemplate.venueId)
+            .all()
+        )
+
+    offers_count_by_venue_id = {}
+    for venue_id in venue_ids:
+        n_individual_offers = individual_offers_count[venue_id] if venue_id in individual_offers_count else 0
```

Comment:
(détail / style) 

```suggestion
        n_individual_offers = individual_offers_count.get(venue_id, 0)
```

idem pour les 2 lignes suivantes.